--- a/iec61850.i
+++ b/iec61850.i
@@ -22,6 +22,31 @@
 {
     return (char *) v;
 }
+FileDirectoryEntry toFileDirectoryEntry(void* data)
+{
+    return (FileDirectoryEntry) data;
+}
+MmsError* MmsError_create()
+{
+    MmsError* error = (MmsError*) malloc(sizeof(MmsError));
+    if (error != NULL)
+        *error = MMS_ERROR_NONE;
+    return error;
+}
+
+int MmsError_getValue(MmsError* self)
+{
+    if (self == NULL)
+        return -1; // Return some error indicator if pointer is NULL
+    
+    return (int)*self;
+}
+
+void MmsErrror_destroy(MmsError* error) {
+    free(error);
+}
+
+
 DataAttribute* toDataAttribute(DataObject * DO)
 { return (DataAttribute*)DO;}
 DataAttribute* toDataAttribute(ModelNode * MN)
@@ -55,6 +80,10 @@
 
 ModelNode* toModelNode(LogicalNode *);
 ModelNode* toModelNode(DataObject *);
+FileDirectoryEntry toFileDirectoryEntry(void*);
+MmsError* MmsError_create();
+void MmsErrror_destroy(MmsError* error);
+int MmsError_getValue(MmsError* self);
 DataAttribute* toDataAttribute(DataObject *);
 DataAttribute* toDataAttribute(ModelNode *);
 DataObject* toDataObject(ModelNode *);
